Cockpit 251 (#449)
cockpit-project/cockpit-project.github.io: _posts/2021-08-18-cockpit-251.md

---
title: Cockpit 251
author: garrett
date: '2021-08-18'
tags: cockpit, machines
slug: cockpit-251
category: release

> summary: ''

Reviewer: Can you please add summary as well?

Author: ```suggestion
summary: 'Improved cerficate usage, better logs filtering, and VM shared directory removal'
```

---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
We release regularly.

Here are the release notes from Cockpit 251 and cockpit-machines 250:

## Certificate/smart card authentication to ssh and sudo

> Once you logged into Cockpit with a certificate, you likely need to switch to administative mode, or connect to remote machines through SSH. If your machine is part of a FreeIPA domain, you can now set up [constrained delegation rules](https://www.freeipa.org/page/V4/Service_Constraint_Delegation) so that the Cockpit certificate authentication gets forwarded to sudo and/or ssh. This is particularly useful if your user account does not have a password. Please see the [Client certification documentation](https://cockpit-project.org/guide/latest/cert-authentication.html) for details.

Author: ```suggestion
For machines in a FreeIPA domain, [constrained delegation rules](https://www.freeipa.org/page/V4/Service_Constraint_Delegation) allow Cockpit to forward certificate authentication to sudo and/or ssh. This enables using certificates and smart cards for signing in, switching administrative mode, and connecting to remote machines through SSH. This is particularly useful for user accounts without passwords. 

Please [read the Client certification documentation](/guide/latest/cert-authentication.html) for details.
```

Reviewer 2: I prefer the original text, with some very very minor tweaks.

Author: I disagree. We're not telling a folksy story here, but that's what the setup is in the old version.

It should get to the point and lead with what this feature is about.

> Once you logged into Cockpit with a certificate, you likely need to switch to administative mode, or connect to remote machines through SSH. If your machine is part of a FreeIPA domain, you can now set up [constrained delegation rules](https://www.freeipa.org/page/V4/Service_Constraint_Delegation) so that the Cockpit certificate authentication gets forwarded to sudo and/or ssh. This is particularly useful if your user account does not have a password. Please see the [Client certification documentation](https://cockpit-project.org/guide/latest/cert-authentication.html) for details.

Reviewer 2: ```suggestion
Once you've logged into Cockpit with a client certificate, you'll likely need to switch to administative mode, or connect to remote machines through SSH. If your machine is part of a FreeIPA domain, you can now set up [constrained delegation rules](https://www.freeipa.org/page/V4/Service_Constraint_Delegation) so that the Cockpit certificate authentication gets forwarded to sudo and/or ssh. This is particularly useful if your user account does not have a password. Please see the [Client certification documentation](https://cockpit-project.org/guide/latest/cert-authentication.html) for details.
```

> ## Logs: Easier to use filtering

Author: ```suggestion
## Logs: Easier-to-use filtering
```

> Logs page now uses a more user friendly filtering options. In previous versions users had to know all keywords to effectively find desired journal messages. The new version contains a dropdown with all possible fields directly spelled out with inline help.

Author: ```suggestion
The Logs page has friendlier filtering options. In previous versions, users had to know esoteric journald keywords to best filter journal entries. Cockpit's new-and-improved log filtering has a rich dropdown with multiple fields and inline help.
```

> Logs page now uses a more user friendly filtering options. In previous versions users had to know all keywords to effectively find desired journal messages. The new version contains a dropdown with all possible fields directly spelled out with inline help.

Reviewer 2: ```suggestion
The Logs page has friendlier filtering options. In previous versions, users had to know esoteric journald keywords to effectively find desired journal messages. The new version contains a dropdown with all possible fields, along with inline help.
```

Middle road?
```
